### PR TITLE
Small pprof application.

### PR DESCRIPTION
**Pull request by davidnewhall**
*Issue #, if available:* N/A

*Description of changes:*

This contribution adds a small app to do `pprof` analysis on this library. This is very small for now, but gives us a base from which to write better tests and whatnot. I'll probably add some more test setups, and I have another contribution that follows this one that adds a work pool to speed up processing time by 2-3x.

The only code changes were to export the `Client` interface from AWS so the external application can mock it (and not actually call AWS services). This is inevitably a no-op.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+list DFS
+list FindLoops
+web mallocgc
+```
+
+The `web` commands opens an SVG file in a web browser. You can save the file
+from your web browser and share it with others. It's often a nice visual.
+
+1.  At the time of this writing, JSON marshaling is the slowest part of this code.
+1.  The second slowest thing is parsing log values into stream and group names.
```
Comment (hossain-rayhan) [inline test/README.md:37]: I might be wrong- should the number be `2`?

**Comment by hossain-rayhan** [comment]
From high level, LGTM. Can you add a screenshot of the performance test result here? Just to see how they look like. Thanks.

File: test/README.md (hunk 1 above)
Comment (davidnewhall) [reply test/README.md:37]: Nope. Numbered list items may always begin with `1.` or `1)`. This makes it easy to re-order items without having to change the number. To test, click the three dots in the top right, and select `View file`.

From https://github.github.com/gfm/#lists:
> The start number of an ordered list is determined by the list number of its initial list item. **The numbers of subsequent list items are disregarded.**

**Comment by hossain-rayhan** [comment]
Hi @davidnewhall are you still interested to work on this? As I know, you had a plan to work another following PRs.

**Comment by davidnewhall** [comment]
I don't mind polishing this up if you're still interested in using it. The project I was working one that used fluent bit was finished and given to another team, so I haven't had much incentive to come back to this. Sorry. I can get you a screenshot though!

**Pull request closed without merging**